### PR TITLE
#159965482 Enables users to update ratings and leave a note after rating.

### PR DESCRIPTION
**Pull request by Mutuba**
## What does this PR do?
```
Implements feedback given during demo.
```

## Description of Task to be completed?
```
Users should be able to rate articles on a scale of 1-5. Users should be able to update their ratings.
Users should be able to give a note that can serve as a reason for their rating.

```

## How should this be manually tested?

```

Clone the repo, cd into ah-leagueOfLegends. Create two accounts. Create articles using HTTP POST on:
http://127.0.0.1:8000/api/articles. Using the other account, rate the article using HTTP POST on: 
http://127.0.0.1:8000/api/articles/your-article-slug/rate/ with the below sample data:
{
  "rate": {
    "rating": 4,
    "note": "I love this article. But changed my mind later"
  }
}

```
## Any background context you want to provide?

```
The rating functionality restricted users to rating only once.

```

## What are the relevant pivotal tracker stories?

[#159965482](https://www.pivotaltracker.com/n/projects/2192365)

## Screenshots if available to support PR.

screen shot 2018-09-26 at 09 20 50

## Questions:

```

Can it be done better?
```

### CODE REVIEW

**Comment by DerKip** [comment]
This is an improvement of this feature. well implemented.

**Review by DerKip — DISMISSED** [review]

**Review by Quantum-35 — DISMISSED** [review]
Looks good to me

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -514,58 +514,65 @@ def post(self, request, slug):
...
                 {'error': 'Rating should be between 1 and 5'},
                 status=status.HTTP_401_UNAUTHORIZED)
-
-        serializer = self.serializer_class(data=rating)
-        serializer.is_valid(raise_exception=True)
-        rating = serializer.data.get('rating')
+        
+        # serializer = self.serializer_class(data=rating)
+        # serializer.is_valid(raise_exception=True)
+        # rating = serializer.data.get('rating')
```
Comment (kevintumbo) [inline authors/apps/articles/views.py]: what is the purpose of keeping this commented out code?

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -514,58 +514,65 @@ def post(self, request, slug):
...
             article = Article.objects.get(slug=slug)
         except Article.DoesNotExist:
             raise NotFound("An article with this slug does not exist")
 
-        current_ratings = ArticleRatings.objects.filter(rater=request.user,
-                                                        article=article).first()
-        article_rating = Article.objects.filter(slug=slug).first()
+        # current_ratings = ArticleRating.objects.filter(rater=request.user,
+        #                                  article=article).first()
+        # article_rating = Article.objects.filter(slug=slug).first()
```
Comment (kevintumbo) [inline authors/apps/articles/views.py]: same as above?

Comment (Mutuba) [reply authors/apps/articles/views.py]: Noted, let me remove it. True, it is not necessary.

**Review by kevintumbo — APPROVED** [review]

**Review by Quantum-35 — APPROVED** [review]

**Pull request merged**